net/utils/net_ipchksum.c

```diff
@@ -170,13 +170,11 @@ uint16_t ipv6_upperlayer_chksum(FAR struct net_driver_s *dev,
 
   /* Sum IP payload data. */
 
-#ifdef CONFIG_MM_IOB
```

Comment:
should we remove the check? since IPBUF also access d_iob too.

Reply:
Done